flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/InputProcessorUtilTest.java

```diff
@@ -58,4 +79,57 @@ public void testGenerateInputGateToChannelIndexOffsetMap() {
 		assertEquals(0, inputGateToChannelIndexOffsetMap.get(ig1).intValue());
 		assertEquals(3, inputGateToChannelIndexOffsetMap.get(ig2).intValue());
 	}
+
+	@Test
+	public void testCreateCheckpointedMultipleInputGate() throws Exception {
+		try (CloseableRegistry registry = new CloseableRegistry()) {
+			MockEnvironment environment = new MockEnvironmentBuilder().build();
+			MockStreamTask streamTask = new MockStreamTaskBuilder(environment).build();
+			StreamConfig streamConfig = new StreamConfig(environment.getJobConfiguration());
+			streamConfig.setCheckpointMode(CheckpointingMode.EXACTLY_ONCE);
+			streamConfig.setUnalignedCheckpointsEnabled(true);
+
+			// First input gate has index larger than the second
+			Collection<IndexedInputGate>[] inputGates = new Collection[] {
+				Collections.singletonList(new MockIndexedInputGate(1, 4)),
+				Collections.singletonList(new MockIndexedInputGate(0, 2)),
+			};
+
+			new MockChannelStateWriter() {
```

Comment:
seems redundant codes `new MockChannelStateWriter()`

Reply:
ops, that's a left over of some previous version.